Upstream: no

From e43ba0857cb39eae30b5e02cb87723e6456dc61e Mon Sep 17 00:00:00 2001
From: Oliver Smith <ollieparanoid@postmarketos.org>
Date: Tue, 10 Nov 2020 20:57:04 +0100
Subject: [PATCH] Revert "layer-shell: error on 0 dimension without anchors"

This reverts commit 8dec751a6d84335fb04288b8efab6dd5c90288d3, which
introduced a consistency check:
https://github.com/swaywm/wlroots/commit/8dec751a6d84335fb04288b8efab6dd5c90288d3

Currently this causes phosh to crash, so let's revert it temporarily:
https://source.puri.sm/Librem5/phosh/-/issues/422
---
 types/wlr_layer_shell_v1.c | 20 --------------------
 1 file changed, 20 deletions(-)

diff --git a/types/wlr_layer_shell_v1.c b/types/wlr_layer_shell_v1.c
index bc681117..d83b22b8 100644
--- a/types/wlr_layer_shell_v1.c
+++ b/types/wlr_layer_shell_v1.c
@@ -307,26 +307,6 @@ static void layer_surface_role_commit(struct wlr_surface *wlr_surface) {
 		return;
 	}
 
-	const uint32_t horiz = ZWLR_LAYER_SURFACE_V1_ANCHOR_LEFT |
-		ZWLR_LAYER_SURFACE_V1_ANCHOR_RIGHT;
-	if (surface->client_pending.desired_width == 0 &&
-		(surface->client_pending.anchor & horiz) != horiz) {
-		wl_resource_post_error(surface->resource,
-			ZWLR_LAYER_SURFACE_V1_ERROR_INVALID_SIZE,
-			"width 0 requested without setting left and right anchors");
-		return;
-	}
-
-	const uint32_t vert = ZWLR_LAYER_SURFACE_V1_ANCHOR_TOP |
-		ZWLR_LAYER_SURFACE_V1_ANCHOR_BOTTOM;
-	if (surface->client_pending.desired_height == 0 &&
-		(surface->client_pending.anchor & vert) != vert) {
-		wl_resource_post_error(surface->resource,
-			ZWLR_LAYER_SURFACE_V1_ERROR_INVALID_SIZE,
-			"height 0 requested without setting top and bottom anchors");
-		return;
-	}
-
 	if (surface->closed) {
 		// Ignore commits after the compositor has closed it
 		return;
-- 
2.20.1

